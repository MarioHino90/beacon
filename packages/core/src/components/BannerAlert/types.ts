import React from "react";
import { ThemedStyledProps } from "styled-components";

import { UntransitionedBannerAlertProps } from "../UntransitionedBannerAlert";

export type Intent = "success" | "warning" | "danger" | "info";

export interface BannerAlertProps
  extends UntransitionedBannerAlertProps,
    Omit<StyledBannerTransitionProps, "children"> {}

export interface StyledBannerTransitionProps
  extends React.HTMLAttributes<HTMLDivElement> {
  isOpen: boolean;
}

export interface IntentContextProviderProps {
  intent: Intent;
}

// this uses <any, any> because the extra props in the themed styled component
// do not matter, just that it is themed
// eslint-disable-next-line @typescript-eslint/no-explicit-any
export interface IntentColorArgs extends ThemedStyledProps<any, any> {
  intent: Intent;
}
