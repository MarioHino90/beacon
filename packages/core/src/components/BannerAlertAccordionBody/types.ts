export interface BannerAlertAccordionBodyProps
  extends React.HTMLAttributes<HTMLDivElement> {}
