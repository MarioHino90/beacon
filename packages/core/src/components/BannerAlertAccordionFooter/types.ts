export interface BannerAlertAccordionFooterProps
  extends React.HTMLAttributes<HTMLDivElement> {}
