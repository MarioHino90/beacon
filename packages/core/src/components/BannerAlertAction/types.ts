import { ButtonProps } from "../Button";
import { Intent } from "../BannerAlert";

export interface BannerAlertActionProps
  extends Omit<ButtonProps, "variant" | "intent"> {
  /** * the visual intent */
  intent?: Intent;
}
