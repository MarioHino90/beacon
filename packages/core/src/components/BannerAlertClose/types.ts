import { Intent } from "../BannerAlert";

export interface BannerAlertCloseProps {
  /**
   * visual banner intents */
  intent: Intent;
  /**
   * on close callack */
  onClose: () => void;
}

export interface StyledProps {
  /**
   * visual banner intents */
  intent: Intent;
}
