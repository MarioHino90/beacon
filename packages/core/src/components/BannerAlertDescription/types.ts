export interface BannerAlertDescriptionProps
  extends React.HTMLAttributes<HTMLDivElement> {}
