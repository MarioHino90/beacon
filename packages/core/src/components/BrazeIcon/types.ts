import React from "react";

export type GenericSpotIcon<Props> = React.LazyExoticComponent<{
  (props: Props): React.ReactElement;
  displayName: string;
}>;

export interface SpotIcons {
  "add-document": GenericSpotIcon<AddDocumentProps>;
  "add-image-or-document": GenericSpotIcon<AddImageOrDocumentProps>;
  "arrow-toggle": GenericSpotIcon<ArrowToggleProps>;
  "campaign-comparison": GenericSpotIcon<CampaignComparisonProps>;
  content: GenericSpotIcon<ContentProps>;
  "email-settings": GenericSpotIcon<EmailSettingsProps>;
  "empty-table": GenericSpotIcon<AddDocumentProps>;
  "funnel-report": GenericSpotIcon<FunnelReportProps>;
  mms: GenericSpotIcon<MMSProps>;
  preview: GenericSpotIcon<PreviewProps>;
  "retention-report": GenericSpotIcon<RetentionReportProps>;
  sms: GenericSpotIcon<SMSProps>;
}

export interface BrazeIconProps extends StyledBrazeIconProps {
  /**
   * icon to be displayed to user */
  name?: keyof SpotIcons;
}

export interface StyledBrazeIconProps
  extends React.HTMLAttributes<HTMLElement> {
  /**
   * child elements to be rendered inside the component tree */
  children?: React.ReactNode | React.ReactNode[];
}

export interface AddDocumentProps
  extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface AddImageOrDocumentProps
  extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface ArrowToggleProps
  extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface CampaignComparisonProps
  extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface ContentProps extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface FunnelReportProps
  extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface EmailSettingsProps
  extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface PreviewProps extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface MMSProps extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface RetentionReportProps
  extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface SMSProps extends React.SVGAttributes<HTMLOrSVGElement> {}
