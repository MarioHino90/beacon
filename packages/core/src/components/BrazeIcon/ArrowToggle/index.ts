export { ArrowToggle } from "./ArrowToggle";

export type { ArrowToggleProps } from "../types";
