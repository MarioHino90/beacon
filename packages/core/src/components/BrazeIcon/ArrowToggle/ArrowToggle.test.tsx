import React from "react";
import { shallow } from "enzyme";
import faker from "@faker-js/faker";

import { ArrowToggleProps } from "../types";

import { ArrowToggle } from "./ArrowToggle";

faker.seed(1604940599873);

describe("ArrowToggle", () => {
  let props: ArrowToggleProps;

  beforeEach(() => {
    props = {
      children: faker.random.words(),
      className: faker.random.word(),
      id: faker.random.word(),
    };
  });

  it("should match the snapshot", () => {
    const wrapper = shallow(<ArrowToggle {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
