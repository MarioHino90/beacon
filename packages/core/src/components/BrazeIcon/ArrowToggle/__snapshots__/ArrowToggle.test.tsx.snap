// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArrowToggle should match the snapshot 1`] = `
<svg
  className="bcl-arrow-toggle-icon Savings"
  fill="none"
  height="24"
  id="input"
  viewBox="0 0 24 24"
  width="24"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    clipPath="url(#clip0)"
  >
    <path
      d="M12 0C5.37258 0 0 5.37258 0 12C0 18.6274 5.37258 24 12 24C18.6274 24 24 18.6274 24 12C24 5.37258 18.6274 0 12 0Z"
      fill="#E9EBED"
    />
    <path
      d="M5.53851 12C5.54674 12.21 5.63319 12.4526 5.77718 12.6058L11.064 18.1862C11.3727 18.552 11.9665 18.5447 12.3136 18.2152C12.6606 17.8855 12.6891 17.3018 12.3397 16.9747L8.4664 12.8812H17.5806C18.0672 12.8812 18.4618 12.4866 18.4618 12C18.4618 11.5133 18.0672 11.1188 17.5806 11.1188H8.4664L12.3397 7.02522C12.6891 6.69807 12.6605 6.11193 12.3132 5.78236C11.9662 5.45279 11.3639 5.45135 11.0638 5.81367L5.777 11.3942C5.61619 11.5651 5.54068 11.7644 5.53833 12H5.53851Z"
      fill="#8F9BA2"
    />
  </g>
  <defs>
    <clipPath
      id="clip0"
    >
      <rect
        fill="white"
        height="24"
        width="24"
      />
    </clipPath>
  </defs>
</svg>
`;
